src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/SamplingPercentageEstimatorSettings.cs

```diff
@@ -0,0 +1,195 @@
+        /// </summary>
+        public double MaxSamplingPercentage { get; set; }
+
+        /// <summary>
+        /// Gets or sets duration of the sampling percentage evaluation 
+        /// interval in seconds.
+        /// </summary>
+        public int EvaluationIntervalSeconds { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating how long to not to decrease
+        /// sampling percentage after last change to prevent excessive fluctuation.
+        /// </summary>
+        public int SamplingPercentageDecreaseTimeoutSeconds { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating how long to not to increase
+        /// sampling percentage after last change to prevent excessive fluctuation.
+        /// </summary>
+        public int SamplingPercentageIncreaseTimeoutSeconds { get; set; }
+
+        /// <summary>
+        /// Gets or sets exponential moving average ratio (factor) applied
+        /// during calculation of rate of telemetry items produced by the application.
+        /// </summary>
+        public double MovingAverageRatio { get; set; }
+
+        /// <summary>
+        /// Gets effective maximum telemetry items rate per second 
+        /// adjusted in case user makes an error while setting a value.
+        /// </summary>
+        internal double EffectiveMaxTelemetryItemsPerSecond
+        {
+            get
+            {
+                return this.MaxTelemetryItemsPerSecond <= 0 ? 1E-12 : this.MaxTelemetryItemsPerSecond;
+            }
+        }
+
+        /// <summary>
+        /// Gets effective initial sampling rate
+        /// adjusted in case user makes an error while setting a value.
+        /// </summary>
+        internal int EffectiveInitialSamplingRate
+        {
+            get
+            {
+                return (int)Math.Floor(100 / AdjustSamplingPercentage(this.InitialSamplingPercentage));
+            }
+        }
+
+        /// <summary>
+        /// Gets effective minimum sampling rate
+        /// adjusted in case user makes an error while setting a value.
+        /// </summary>
+        internal int EffectiveMinSamplingRate
+        {
+            get
+            {
+                return (int)Math.Floor(100 / AdjustSamplingPercentage(this.MaxSamplingPercentage));
+            }
+        }
+
+        /// <summary>
+        /// Gets effective maximum sampling rate
+        /// adjusted in case user makes an error while setting a value.
+        /// </summary>
+        internal int EffectiveMaxSamplingRate
+        {
+            get
+            {
+                return (int)Math.Ceiling(100 / AdjustSamplingPercentage(this.MinSamplingPercentage));
+            }
+        }
+
+        /// <summary>
+        /// Gets effective sampling percentage evaluation interval
+        /// adjusted in case user makes an error while setting a value.
+        /// </summary>
+        internal int EffectiveEvaluationIntervalSeconds
```

Comment:
Changed in the "settings" class. Since it is not generally used in config file (I was informed there are problems in deserializing complex types like TimeSpan or DateTime, etc), I left telemetry processor parameters as "...seconds"